__tests__/__unit__/job/actions.unit.test.ts

```diff
@@ -0,0 +1,508 @@
+} from "../../../__mocks__/generators/shared";
+import { generateIJobObject, generateJobsTree } from "../../../__mocks__/generators/jobs";
+import { generateJesApi, bindJesApi } from "../../../__mocks__/generators/api";
+import * as jobActions from "../../../src/job/actions";
+import { ZoweDatasetNode } from "../../../src/dataset/ZoweDatasetNode";
+import * as dsActions from "../../../src/dataset/actions";
+import * as globals from "../../../src/globals";
+import { generateDatasetSessionNode, generateDatasetTree } from "../../../__mocks__/generators/datasets";
+import { Profiles } from "../../../src/Profiles";
+
+const activeTextEditorDocument = jest.fn();
+
+function createGlobalMocks() {
+    Object.defineProperty(vscode.window, "showInformationMessage", { value: jest.fn(), configurable: true });
+    Object.defineProperty(vscode.window, "showInputBox", { value: jest.fn(), configurable: true });
+    Object.defineProperty(vscode.window, "showErrorMessage", { value: jest.fn(), configurable: true });
+    Object.defineProperty(zowe, "IssueCommand", { value: jest.fn(), configurable: true });
+    Object.defineProperty(zowe.IssueCommand, "issueSimple", { value: jest.fn(), configurable: true });
+    Object.defineProperty(vscode.window, "showOpenDialog", { value: jest.fn(), configurable: true });
+    Object.defineProperty(zowe, "GetJobs", { value: jest.fn(), configurable: true });
+    Object.defineProperty(zowe.GetJobs, "getJclForJob", { value: jest.fn(), configurable: true });
+    Object.defineProperty(vscode.workspace, "openTextDocument", { value: jest.fn(), configurable: true });
+    Object.defineProperty(vscode.window, "showTextDocument", { value: jest.fn(), configurable: true });
+    Object.defineProperty(zowe, "ZosmfSession", { value: jest.fn(), configurable: true });
+    Object.defineProperty(zowe.ZosmfSession, "createBasicZosmfSession", { value: jest.fn(), configurable: true });
+    Object.defineProperty(vscode.window, "activeTextEditor", { value: jest.fn(), configurable: true });
+    Object.defineProperty(vscode.window, "showQuickPick", { value: jest.fn(), configurable: true });
+    Object.defineProperty(vscode.window.activeTextEditor, "document", {
+        get: activeTextEditorDocument,
+        configurable: true
+    });
+    Object.defineProperty(globals, "LOG", { value: jest.fn(), configurable: true });
+    Object.defineProperty(globals.LOG, "debug", { value: jest.fn(), configurable: true });
+    Object.defineProperty(globals.LOG, "error", { value: jest.fn(), configurable: true });
+    Object.defineProperty(Profiles, "getInstance", { value: jest.fn(), configurable: true });
+}
+
+// Idea is borrowed from: https://github.com/kulshekhar/ts-jest/blob/master/src/util/testing.ts
+const mocked = <T extends (...args: any[]) => any>(fn: T): jest.Mock<ReturnType<T>> => fn as any;
+
+describe("Jobs Actions Unit Tests - Function setPrefix", () => {
+    let blockMocks;
```

Comment:
Same as in previous PR, this should become a `const` and go into each `it` block. We will deal with it in next PR after we get all the current refactoring PRs in.